aurora/src/test/java/com/market/banica/aurora/AuroraComponentIT.java

```diff
@@ -0,0 +1,865 @@
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import static com.aurora.AuroraServiceGrpc.newStub;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+@SpringJUnitConfig
+@SpringBootTest(classes = AuroraApplication.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("testAuroraIT")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+class AuroraComponentIT {
+
+
+    @Autowired
+    private ChannelManager channelManager;
+
+    @Autowired
+    private Publishers publishers;
+
+    @Autowired
+    private JMXConfig jmxConfig;
+
+    @Autowired
+    private AuroraServiceImpl auroraService;
+
+    @Rule
+    public GrpcCleanupRule grpcCleanup = new GrpcCleanupRule();
+
+    @LocalServerPort
+    private int grpcMockPort;
+
+    private String receiverChannelName;
+    private ManagedChannel receiverChannel;
+
+    private String senderName;
+    private ManagedChannel senderChannel;
+
+    private AuroraServiceGrpc.AuroraServiceBlockingStub blockingStub;
+    private AuroraServiceGrpc.AuroraServiceStub stub;
+
+    private static String channelsBackupUrl;
+    private static String publishersBackupUrl;
+
+    private static final Set<String> channelNames = new HashSet<>();
+
+    private static String currentPublisher;
+
+
+    @BeforeAll
+    public static void getFilePath(@Value("${aurora.channels.file.name}") String channels, @Value("${aurora.channels.publishers}") String publishersFileName) {
+        channelsBackupUrl = channels;
+        publishersBackupUrl = publishersFileName;
+
+    }
+
+
+    @AfterAll
+
+    public static void cleanUp() throws IOException {
+        ApplicationDirectoryUtil.getConfigFile(channelsBackupUrl).delete();
+        ApplicationDirectoryUtil.getConfigFile(publishersBackupUrl).delete();
+    }
+
+
+    @BeforeEach
+    public void setup() {
+        senderName = InProcessServerBuilder.generateName();
+        senderChannel = InProcessChannelBuilder
+                .forName(senderName)
+                .directExecutor()
+                .build();
+
+        receiverChannelName = InProcessServerBuilder.generateName();
+        receiverChannel = InProcessChannelBuilder
+                .forName(receiverChannelName)
+                .executor(Executors.newSingleThreadExecutor()).build();
+
+        channelNames.add(receiverChannelName);
+
+    }
+
+    @AfterEach
+    public void tearDown() {
+        publishers.deletePublisher(currentPublisher);
+        channelManager.removeChannel(currentPublisher);
+    }
+
+
+    @Test
+    void request_Should_ForwardToReceiverResponse() throws IOException {
```

Comment:
All test names are not clear enough. Condition is missing in them. In this case it is not clear enough why/when "request should be forward to receiver response".

This applies for most of the other tests.

Reply:
fixed.